src/main/java/com/thealgorithms/backtracking/AllPathsFromSourceToTarget.java

```diff
@@ -0,0 +1,120 @@
+/** Author : Siddhant Swarup Mallick
+ * Github : https://github.com/siddhant2002
+ */
+
+/** Program description - To find all possible paths from source to destination*/
+
+/**Wikipedia link -> https://en.wikipedia.org/wiki/Shortest_path_problem */
+package com.thealgorithms.backtracking;
+
+// JAVA program to print all paths from a source to destination.
+import java.util.*;
+
+// A directed graph using adjacency list representation
+
+
+public class AllPathsFromSourceToTarget {
+
+    // No. of vertices in graph
+    private int v;
+
+    // adjacency list
+    private ArrayList<Integer>[] adjList;
+
+    // Constructor
+    public AllPathsFromSourceToTarget(int vertices)
+    {
+
+        // initialise vertex count
+        this.v = vertices;
+
+        // initialise adjacency list
+        initAdjList();
+    }
+
+    // utility method to initialise adjacency list
+    
+    @SuppressWarnings("unchecked")
```

Comment:
Is this necessary? Please fix the warning without suppressing it

Reply:
> Is this necessary? Please fix the warning without suppressing it

No, it's not necessary